components/builder-api-client/src/builder.rs

```diff
@@ -399,28 +409,27 @@ impl BuilderAPIProvider for BuilderAPIClient {
                                    token: &str,
                                    promote: bool)
                                    -> Result<()> {
+        debug!("Promote/demote for group: {}, channel: {}",
+               group_id, channel);
+
         let json_idents = json!(idents);
         let body = json!({ "idents": json_idents });
-        let sbody = serde_json::to_string(&body).unwrap();
+
         let url = format!("jobs/group/{}/{}/{}",
                           group_id,
                           if promote { "promote" } else { "demote" },
                           channel);
-        let res = self.add_authz(self.0.post(&url), token)
-                      .body(&sbody)
-                      .header(Accept::json())
-                      .header(ContentType::json())
-                      .send()
-                      .map_err(Error::HyperError)?;
-
-        if res.status != StatusCode::NoContent {
-            debug!("Failed to {} group, status: {:?}",
-                   if promote { "promote" } else { "demote" },
-                   res.status);
-            return Err(err_from_response(res));
-        }
 
-        Ok(())
+        let resp = self.add_authz(self.0.post(&url), token)
+                       .json(&body)
+                       .send()?;
+        debug!("Status: {:?}", resp.status());
```

Comment:
Going back to my comment above re: structured logging, will logging _just_ the status here, absent any other information about a specific request, be useful?

Reply:
It's semi useful atm... I did some work to make the logging a bit cleaner and consistent but that definitely hasn't been the main thrust of this PR...